packages/template-ui/src/views/ListSection.vue

```diff
@@ -43,16 +43,57 @@ import { mapState, mapGetters } from 'vuex';
 
 export default {
   name: 'ListSection',
+  props: {
+    section: Object,
+    sectionNodes: {
+      type: Array,
+      default() {
+        return [];
+      },
+    },
+    // FIXME use the loading prop:
+    // loading: Boolean,
+  },
+  data() {
+    return {
+      subsectionNodes: {},
+      loadingSubsectionNodes: true,
+    };
+  },
   computed: {
-    ...mapState(['section', 'cardColumns', 'mediaQuality', 'hasFilters']),
+    ...mapState(['cardColumns', 'mediaQuality', 'hasFilters']),
     ...mapGetters({
-      isInlineLevel: 'isInlineLevel',
       getAssetURL: 'getAssetURL',
       isFilterEmpty: 'filters/isEmpty',
     }),
+    isInlineLevel() {
+      return this.sectionNodes.every((n) => n.kind === 'topic');
+    },
     backgroundImageURL() {
       return this.getAssetURL('sectionBackgroundImage');
     },
   },
+  watch: {
+    sectionNodes() {
+      if (!this.isInlineLevel) {
+        return;
+      }
+      return this.fetchSubsectionNodes();
+    },
+  },
+  methods: {
+    fetchSubsectionNodes() {
+      this.loadingSubsectionNodes = true;
+      this.subsectionNodes = {};
+      return Promise.all(this.sectionNodes.map((subsection) => {
+        return window.kolibri.getContentByFilter({ parent: subsection.id })
+          .then((page) => {
+            this.subsectionNodes[subsection.id] = page.results;
+          });
+      })).then(() => {
+        this.loadingSubsectionNodes = false;
+      });
```

Comment:
Maybe the second `then` is not needed here, we can add the `this.loadingSubsectionNodes = false` inside the first one.

Reply:
I've just noticed that this is a `Promise.all`, so forget about this comment.